rpc_state_reader_sn_api/src/lib.rs

```diff
@@ -1030,3 +1096,217 @@ mod blockifier_transaction_tests {
         }
     }
 }
+
+mod starknet_in_rust_transaction_tests {
+    use cairo_vm::felt::{felt_str, Felt252};
+    use starknet_api::{core::PatriciaKey, stark_felt, transaction::TransactionHash};
+    use starknet_in_rust::{
+        core::errors::state_errors::StateError,
+        definitions::{
+            block_context::{BlockContext, StarknetChainId, StarknetOsConfig},
+            constants::{
+                DEFAULT_CAIRO_RESOURCE_FEE_WEIGHTS,
+                DEFAULT_CONTRACT_STORAGE_COMMITMENT_TREE_HEIGHT,
+                DEFAULT_GLOBAL_STATE_COMMITMENT_TREE_HEIGHT, DEFAULT_INVOKE_TX_MAX_N_STEPS,
+                DEFAULT_VALIDATE_MAX_N_STEPS,
+            },
+        },
+        execution::TransactionExecutionInfo,
+        services::api::contract_classes::compiled_class::CompiledClass,
+        state::{
+            cached_state::{CachedState, ContractClassCache},
+            state_api::StateReader,
+            state_cache::StorageEntry,
+            BlockInfo,
+        },
+        transaction::{InvokeFunction, Transaction},
+        utils::{Address, ClassHash},
+    };
+
+    use super::*;
+
+    pub struct RpcStateReader(RpcState);
+
+    impl StateReader for RpcStateReader {
+        fn get_contract_class(&self, class_hash: &ClassHash) -> Result<CompiledClass, StateError> {
+            let hash = ClassHash(StarkHash::new(*class_hash).unwrap());
+            Ok(CompiledClass::from(self.0.get_contract_class(&hash)))
+        }
+
+        fn get_class_hash_at(&self, contract_address: &Address) -> Result<ClassHash, StateError> {
+            let address = ContractAddress(
+                PatriciaKey::try_from(
+                    StarkHash::new(contract_address.clone().0.to_be_bytes()).unwrap(),
```

Comment:
I'm not sure we need to clone the addresses in these methods.

Reply:
We don't. It has been removed.